File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set nums_set(nums.begin(), nums.end());
        int max_len = 0;
        for(const auto& number : nums)
        {
            if(nums_set.contains(number-1)) continue;
            int len = 0;
            while(nums_set.contains(number+len)) len++;
            if (len > max_len) max_len = len;
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-2.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r= 0, len(nums) - 1
        while l <= r:
            c = (l + r) // 2
            if nums[c] == target:
                return c
            elif nums[c] < target:
                memory = c + 1
                l = c + 1
            else:
                memory = c
                r = c - 1
        return memory